velox/functions/prestosql/window/NthValue.cpp

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/common/base/Exceptions.h"
+#include "velox/expression/FunctionSignature.h"
+#include "velox/functions/lib/window/NthValueBase.h"
+#include "velox/vector/FlatVector.h"
+
+namespace facebook::velox::window::prestosql {
+
+namespace {
+class NthValue : public functions::window::NthValueBase {
+ public:
+  NthValue(
+      const std::vector<exec::WindowFunctionArg>& args,
+      const TypePtr& resultType,
+      velox::memory::MemoryPool* pool)
+      : NthValueBase(args, resultType, pool) {
+    if (args[1].constantValue) {
+      if (args[1].constantValue->isNullAt(0)) {
+        initializeConstantOffset(std::nullopt);
+      } else {
+        initializeConstantOffset(std::make_optional(
+            args[1]
+                .constantValue->template as<ConstantVector<int64_t>>()
+                ->valueAt(0)));
+      }
+    } else {
+      initializeOffsetVector(args[1], pool);
+    }
+  }
+};
+} // namespace
+
+// nth_value(T, bigint) -> T
```

Comment:
Please move this line within the function above the signatures initialization.

Also, add a full stop to the end.